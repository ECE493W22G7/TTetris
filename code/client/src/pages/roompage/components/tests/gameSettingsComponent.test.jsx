import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import GameSettingsComponent from '../gameSettingsComponent';
import SocketMock  from 'socket.io-mock';
import { toMatchDiffSnapshot } from 'snapshot-diff';
const {act} = renderer;


expect.extend({ toMatchDiffSnapshot });

function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

describe("GameSettingsComponent", () => {
    let socket;
    let container;

    beforeEach(() => {
        socket = new SocketMock();
            socket.on("getConnectedClients", (roomId) => {
            console.log("testasdfsf");
        })
    });



    
    test('renders correctly', async () => {
            let component;
            act(() => {
                component = renderer.create(<GameSettingsComponent socket={socket.socketClient} roomID={"34223"} />);
            });
            
            await sleep(2000);
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
    });

     test("should send 'changeGameSpeed' to socket when the slider is moved", (done) => {
         socket.on("changeGameSpeed",  (roomId, newSpeed) =>{
            expect(newSpeed).toEqual("10");
            done()
         });

        render(<GameSettingsComponent socket={socket.socketClient} roomID={"34223"} />);
        const slider = document.querySelector('[data-testid="speed_input"]');
        fireEvent.change(slider, { target: { value: 10 } });
     })

    test("should send 'changeClientName' to socket when the nickname is changed", (done) => {
         socket.on("changeClientName",  (roomId, clientId, newName) =>{
            expect(newName).toEqual("Lorem");
            done()
         });

        render(<GameSettingsComponent socket={socket.socketClient} roomID={"34223"} />);
        
        userEvent.paste(screen.getByTestId("nickname_input"), "Lorem");
     })

    test("should send 'changeClientColor' to socket when the color is changed", async () => {
         socket.on("changeGameSpeed",  (roomId, clientID, newColor) =>{
            expect(newColor).toEqual("#0000ff");
            done()
         });

        render(<GameSettingsComponent socket={socket.socketClient} roomID={"34223"} />);
        const colorSelector = document.querySelector('[data-testid="color_input"]');
        fireEvent.change(colorSelector, { target: { value: "#0000ff" } });
     })

})

