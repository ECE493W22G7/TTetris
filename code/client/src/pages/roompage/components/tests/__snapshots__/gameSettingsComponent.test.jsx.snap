// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameSettingsComponent renders correctly 1`] = `
<div
  className="game-sections card"
>
  <div
    className="game-settings-component card"
  >
    <div
      className="settings-content"
    >
      <span
        className="game-title font-italic text-center"
      >
        Host Settings
      </span>
      <div
        className="host-input"
      >
        <div
          className="input-group"
        >
          <span
            className="input-group-text"
            id="inputGroup-sizing-default"
          >
            Set Speed:
          </span>
          <input
            className="form-control set-speed"
            data-testid="speed_input"
            disabled={true}
            max="15"
            min="1"
            onChange={[Function]}
            step="1"
            type="range"
            value={1}
          />
        </div>
        <button
          className="start-button btn btn-primary"
          disabled={true}
          onClick={[Function]}
          role="button"
          type="button"
        >
          Start Game
        </button>
      </div>
    </div>
  </div>
  <div
    className="game-settings-component card"
  >
    <div
      className="settings-content"
    >
      <span
        className="game-title font-italic text-center"
      >
        Player Settings
      </span>
      <div
        className="player-input"
      >
        <div
          className="input-group mb-3"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="inputGroup-sizing-default"
            >
              Nickname:
            </span>
          </div>
          <input
            aria-describedby="inputGroup-sizing-default"
            aria-label="Default"
            className="form-control"
            data-testid="nickname_input"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="player-input"
      >
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              id="inputGroup-sizing-default"
            >
              Piece Color:
            </span>
          </div>
          <input
            aria-describedby="inputGroup-sizing-default"
            aria-label="Default"
            className="form-control"
            data-testid="color_input"
            onChange={[Function]}
            style={
              Object {
                "height": "40px",
              }
            }
            type="color"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
